src/input/keyboard/mod.rs

```diff
@@ -477,13 +480,15 @@ pub struct XkbContext<'a> {
 
 impl<'a> XkbContext<'a> {
     /// Get the reference to the xkb state.
-    pub fn xkb(&self) -> &Xkb {
+    pub fn xkb(&self) -> &Mutex<Xkb> {
```

Comment:
Would it be cleaner to already return a `MutexGuard`?

Reply:
I'm not sure if there's any general convention for this sort of thing. If so maybe name it `.lock_xkb()` so it's visible that this is holding onto a lock of something.

Reply:
I don't mind either way too much to be honest. You can't do anything nasty with a read-only reference of a Mutex compared to a MutexGuard anyway.

Reply:
Then I think it can be left returning a `&Mutex<Xkb>`. I think that's good for now.